source/module_esolver/esolver_ks_lcao.cpp

```diff
@@ -1096,19 +1096,24 @@ template <typename TK, typename TR>
 void ESolver_KS_LCAO<TK, TR>::after_scf(const int istep)
 {
     ModuleBase::TITLE("ESolver_KS_LCAO", "after_scf");
-
-    // 1) call after_scf() of ESolver_KS
+    // 1) calculate the kinetic energy density tau, sunliang 2024-09-18
+    if (PARAM.inp.out_elf)
+    {
+        this->pelec->cal_tau(*(this->psi));
```

Comment:
cal_tau() is te same in all esolvers, thus should be put into after_scf() of esolver_fp.

Reply:
Since `psi` is defined in `esolver_ks`, `cal_tau(psi)` may be put into `esolver_ks`. However, after doing so, a segmentation fault occurred, and I am unsure of the cause, so maybe we can keep them in `esolver_ks_pw` amd `esolver_ks_lcao`.